estuary/utils/story.py

```diff
@@ -195,9 +205,133 @@ def get_wait_times(self, results):
+            # the processing in a FreshmakerEvent, so we do not count it towards the total wait time
+            if artifact.__label__ != 'FreshmakerEvent':
+                total_wait_time += wait_time.total_seconds()
+
+        return wait_times, total_wait_time
+
+    def get_total_processing_time(self, results):
+        """
+        Get the total time spent processing in the story.
+
+        :param list results: contains inflated results from Neo4j
+        :return: the seconds of total time spent processing
+        :rtype: int
+        """
+        total = 0
+        build = None
+        timed_processes = {
+            'FreshmakerEvent': ['time_created', 'time_done'],
+            'KojiBuild': ['creation_time', 'completion_time'],
+            'ModuleKojiBuild': ['creation_time', 'completion_time'],
+            'ContainerKojiBuild': ['creation_time', 'completion_time'],
+            'Advisory': ['created_at', 'status_time'],
+            'ContainerAdvisory': ['created_at', 'status_time']
+        }
+        for index, artifact in enumerate(results):
+            if artifact.__label__ not in timed_processes:
+                continue
+
+            creation_time = getattr(artifact, timed_processes[artifact.__label__][0])
+            if not creation_time:
+                id_num = getattr(artifact, artifact.unique_id_property + '_')
+                log.warning(
+                    'While calculating the total processing time, a %s with id %s was encountered '
+                    'without a creation time.',
+                    artifact.__label__, id_num
+                )
+                continue
+
+            if artifact.__label__ in ['KojiBuild', 'ModuleKojiBuild', 'ContainerKojiBuild']:
+                build = artifact
+
+            if artifact.__label__ in ['Advisory', 'ContainerAdvisory']:
+                if artifact.state not in ['SHIPPED_LIVE', 'DROPPED_NO_SHIP']:
+                    completion_time = datetime.utcnow()
+                else:
+                    completion_time = getattr(artifact, timed_processes[artifact.__label__][1])
+                if build:
+                    id_num = getattr(build, 'id_')
+                    creation_time = artifact.attached_build_times[id_num]
+
+            # We do not want the processing time of the entire FreshmakerEvent, just the
+            # processing time until the displayed ContainerKojiBuild is created
+            elif artifact.__label__ == 'FreshmakerEvent':
+                if index != len(results) - 1:
+                    next_artifact = results[index + 1]
+                    completion_time = getattr(next_artifact,
+                                              timed_processes[next_artifact.__label__][0])
+                elif artifact.state_name in ['COMPLETE', 'SKIPPED', 'FAILED', 'CANCELED']:
+                    completion_time = getattr(artifact, timed_processes['FreshmakerEvent'][1])
```

Comment:
What if this is a legacy `FreshmakerEvent` where `time_done` is `None`?

Reply:
Do you mean if it was stubbed out? Would a log.wanring be needed then?